### DISCUSSION TITLE
Visible mesh geometry indices in Fresnel Renderings

### DISCUSSION THREAD

**Discussion by raniatze** [glotzerlab/fresnel#237] [Support]
I'm working with a Fresnel scene in which I have added multiple mesh geometries using fresnel.geometry.Mesh. I have successfully rendered Bird's Eye View (BEV) images of the scene using fresnel.tracer.Path. My goal is to determine the indices of the mesh geometries that are visible in the rendered images.

I attempted to assign a unique color to each mesh geometry and identify the indices based on these colors in the rendered image, but the results have not been reliable. Is there a more accurate method to get the indices of the mesh geometries that are present in the BEV rendering?

**Comment by joaander** [comment]
One could add a new type of tracer that outputs the mesh index to a buffer instead of colors for a clean API. However, a solid color is as good as a 3 dimensional index (with resolution 0-255 in each dimension).

You don't say what is not reliable about your implementation. I would expect a solid material with a specific color to be passed through without any modifications to the output in a `Preview` tracer. You should also disable anti-aliasing to prevent any mixed colors at shape edges.
